### PR TITLE
feat(frontend): 管理画面に提出状況詳細画面を実装

### LINKED ISSUE #30: 管理画面に提出状況詳細画面を実装する
各生徒の解答の詳細を確認し、コードやテスト結果を閲覧できる画面。再提出やフィードバックを行うことができる。

### PR DESCRIPTION
**Pull request by Texiko-texiko**
close #30 

This pull request introduces a new route for viewing submission details in the admin section of the application. It includes changes to the route tree and the implementation of new components to display submission information, submitted code, and test results.

### Route Tree Updates:
* Added a new route `AdminSubmissionsSubmissionIdRoute` to the route tree (`frontend/src/route-tree.gen.ts`). [[1]](diffhunk://#diff-48bb38e44fc5b79b1be126f1f4580bafaedd2188b5af99a54412f99157d572feR21) [[2]](diffhunk://#diff-48bb38e44fc5b79b1be126f1f4580bafaedd2188b5af99a54412f99157d572feR53-R63) [[3]](diffhunk://#diff-48bb38e44fc5b79b1be126f1f4580bafaedd2188b5af99a54412f99157d572feR100-R106) [[4]](diffhunk://#diff-48bb38e44fc5b79b1be126f1f4580bafaedd2188b5af99a54412f99157d572feR123-R131) [[5]](diffhunk://#diff-48bb38e44fc5b79b1be126f1f4580bafaedd2188b5af99a54412f99157d572feR140) [[6]](diffhunk://#diff-48bb38e44fc5b79b1be126f1f4580bafaedd2188b5af99a54412f99157d572feR150-R164) [[7]](diffhunk://#diff-48bb38e44fc5b79b1be126f1f4580bafaedd2188b5af99a54412f99157d572feR173-R182) [[8]](diffhunk://#diff-48bb38e44fc5b79b1be126f1f4580bafaedd2188b5af99a54412f99157d572feR199) [[9]](diffhunk://#diff-48bb38e44fc5b79b1be126f1f4580bafaedd2188b5af99a54412f99157d572feR212-R214)

### New Components:
* Created `SubmissionInfo` component to display basic submission details (`frontend/src/routes/admin/submissions/$submissionId/-components/submission-info/submission-info.tsx`).
* Created `SubmissionedCode` component to display the submitted code with a copy-to-clipboard feature (`frontend/src/routes/admin/submissions/$submissionId/-components/submissioned-code/submissioned-code.tsx`).
* Created `TestResults` component to display the test results of the submission (`frontend/src/routes/admin/submissions/$submissionId/-components/test-results/test-results.tsx`).

### Component Exports:
* Added export statements for newly created components (`frontend/src/routes/admin/submissions/$submissionId/-components/submission-info/index.ts`, `frontend/src/routes/admin/submissions/$submissionId/-components/submissioned-code/index.ts`, `frontend/src/routes/admin/submissions/$submissionId/-components/test-results/index.tsx`). [[1]](diffhunk://#diff-fd69cdab00b328d69cedcac7d0c438ce4329d2cdd91adcd2d846495a80b9d52bR1) [[2]](diffhunk://#diff-bc94494d7e93c948e5b7e4f8cf0dce2b928dfd5dad2c71c8d978cfc2222379fdR1) [[3]](diffhunk://#diff-527534cfdc8b48afbece4e8a683b01227ef52f72ea58613faa35ef38ca95edefR1)

### Route Implementation:
* Implemented the route handler for the new submission detail route (`frontend/src/routes/admin/submissions/$submissionId/route.tsx`, `frontend/src/routes/admin/submissions/$submissionId/route.lazy.tsx`). [[1]](diffhunk://#diff-06868f6ca6013993bd4850073e5b3c6927338243cfea7caec74c4e949d6a6198R1-R21) [[2]](diffhunk://#diff-52951944277054c5b9f62651baf465164ba0986adbbffe0d963cd6b6faa7aee0R1-R50)

### CODE REVIEW

File: frontend/src/routes/admin/submissions/$submissionId/-components/submissioned-code/submissioned-code.tsx
Comment (r4ai) [inline frontend/src/routes/admin/submissions/$submissionId/-components/submissioned-code/submissioned-code.tsx:1]: imo: submissionedという英語は存在しないので、submitted-code-infoやsubmitted-codeなどに名前を変えた方が分かりやすいと思います！

File: frontend/src/routes/admin/submissions/$submissionId/route.lazy.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        <Link href="..">
+          <Button size="sm" variant="outline">
+            <ArrowLeft className="mr-2 h-4 w-4" />
+            提出一覧に戻る
+          </Button>
+        </Link>
+      </div>
+      <h1 className="mb-6 text-2xl font-bold">提出詳細</h1>
+      <div className="flex flex-1 gap-6">
+        <div className="flex flex-1 flex-col">
```
Comment (r4ai) [inline frontend/src/routes/admin/submissions/$submissionId/route.lazy.tsx:35]: imo: 左右のカードの間にも余白を入れてあげた方が見やすいかなと思いました！

```suggestion
        <div className="flex flex-1 flex-col gap-6">
```

File: frontend/src/routes/admin/submissions/$submissionId/-components/submissioned-code/submissioned-code.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        </div>
+      </CardHeader>
+      <CardContent className="flex-1 overflow-auto">
+        <div className="h-full overflow-x-auto rounded-md bg-gray-100 p-4">
+          <ReadOnlyCodeBlock
+            code={submission.code}
+            language={submission.language.name.toLowerCase()}
+          />
+        </div>
+      </CardContent>
```
Comment (r4ai) [inline frontend/src/routes/admin/submissions/$submissionId/-components/submissioned-code/submissioned-code.tsx:52]: imho: コピーボタンの位置が少し崩れているように感じたので、コードブロックの中の右上などに置いた方が分かりやすいような気がしました？人によって違う気がするのでこれについて意見を聞きたいです

image

**Review by r4ai — CHANGES_REQUESTED** [review]

**Review by r4ai — APPROVED** [review]
LGTM!

**Review by sou1118 — DISMISSED** [review]

**Review by sou1118 — APPROVED** [review]
LGTM!

**Review by r4ai — APPROVED** [review]

**Pull request merged**